renku/core/models/template.py

```diff
@@ -60,19 +60,26 @@ def templates(self) -> List["Template"]:
 
         return self.manifest.templates
 
-    @abstractmethod
     def is_update_available(self, id: str, reference: Optional[str], version: Optional[str]) -> Tuple[bool, str]:
-        """Return True if an update is available along with the latest version of a template."""
-        raise NotImplementedError
+        """Return True if an update is available along with the latest reference of a template."""
+        latest = self.get_latest_reference_and_version(id=id, reference=reference, version=version)
+        if not latest:
+            return False, reference
+
+        latest_reference, latest_version = latest
+        update_available = latest_reference != reference or latest_version != version
+        return update_available, latest_reference
```

Comment:
This would mean that if I install hypothetical version 1.3.0 and migrate, and then install hypothetical version 1.2.0 and run migration check, it'd say a new version is available and downgrade, right?

Reply:
No, `EmbeddedTemplates::get_latest_reference_and_version` checks if project's template version is newer than renku's template version and returns the latest of the two, so, `1.3.0` will be the latest version/reference in that case and nothing will be changed.